replication-api-impl/src/main/java/com/connexta/replication/api/impl/worker/Worker.java

```diff
@@ -0,0 +1,335 @@
+    if (!OperationType.UNKNOWN.equals(task.getOperation())) {
+      return false;
+    }
+
+    List<MetadataInfo> infos = task.metadatas().collect(Collectors.toList());
+    if (infos.size() > 1) {
+      LOGGER.warn("More than one metadata is currently not supported by this worker");
+      return false;
+    }
+
+    DDFMetadataInfo ddfInfo = infos.stream().filter(DDFMetadataInfo.class::isInstance)
+        .map(DDFMetadataInfo.class::cast)
+        .findFirst()
+        .orElse(null);
+
+    if (ddfInfo == null) {
+      LOGGER.info("Unable to handle any of the metadatas received in this task");
+      return false;
+    }
+
+    return true;
+  }
+
+  private void handle(Task task) throws InterruptedException {
+    NodeAdapter destination = getDestinationAdapter(task);
+    if (destination == null) {
+      task.unlock();
+      return;
+    }
+
+    if (!adaptersAvailable(task, destination)) {
+      return;
+    }
+
+    List<DDFMetadataInfo> infos = task.metadatas()
+        .filter(DDFMetadataInfo.class::isInstance)
+        .map(DDFMetadataInfo.class::cast)
+        .collect(Collectors.toList());
+
+    final DDFMetadataInfo info = infos.get(0);
+    final Metadata metadata = new MetadataImpl(((Map) info.getData()).get("raw-metadata"),
+        info.getDataClass(), task.getId(),
+        Date.from(info.getLastModified()));
+
+    final OperationType type = task.getOperation();
+    switch (type) {
+      case HARVEST:
+      case REPLICATE_FROM:
+        doCreate(task, metadata, destination, local);
+        break;
+      case REPLICATE_TO:
+        doCreate(task, metadata, local, destination);
+        break;
+      case DELETE:
+        doDelete(task, metadata, destination);
+        break;
+      default:
+        // should never hit this since deserialization will return UNKNOWN
+        task.unlock();
+    }
+  }
+
+  @Nullable
+  private NodeAdapter getDestinationAdapter(Task task) {
+    Site site;
+    final String siteId = task.getQueue().getSite();
+    try {
+      site = siteManager.get(siteId);
+    } catch (NotFoundException e) {
+      LOGGER.debug("Unable to find site {} for queue", siteId, e);
+      return null;
+    } catch (ReplicationPersistenceException e) {
+      LOGGER.debug("Failed to fetch site {} for queue", siteId, e);
+      return null;
+    }
+
+    return nodeAdapters.factoryFor(site.getType())
+        .create(site.getUrl());
+  }
+
+  private void doCreate(final Task task, final Metadata metadata, final NodeAdapter source,
+      final NodeAdapter destination)
+      throws InterruptedException {
+    final String metadataId = task.getId();
+    final String operation = task.getOperation().name();
+
+    try {
+      // TODO: remove existence check once better reporting is done by adapters
+      // see https://github.com/connexta/replication/issues/235
+      if (source.exists(metadata)) {
+        CreateRequest request = new CreateRequestImpl(List.of(metadata));
```

Comment:
❗️ Doing separate metadata and resource creates will not work with any DDF based system. It doesn't really work with Ion either but it the adapter just quietly fails for metadata creates. Ion only supports resource+metadata and DDF only supports metadata only, or both resource + metadata in one request.

Reply:
I know this concept was talked about a little bit but we are not ready for that yet. We can always add this functionality later if needed and the systems can support it.

Reply:
Okay, I can change this to check if a resource exists first, and if it does do the resource create, otherwise do a metadata create.

Reply:
Or aren't we always only harvesting from DDF based systems? @paouelle

Reply:
at the moment, we are only harvesting from regional DDF systems as far as I know. This should be ensured by the query service.

Reply:
Actually based on our discussion yesterday, if we want to make DDF the local site in order to continue support for the E2E tests, I will have to make this change.

Reply:
Updated.